### PR TITLE
Support Kfiou Loss

### PR DESCRIPTION
**Pull request by OrangeSodahub**
### rotated_retinanet_hbb_kfiou_r50_fpn_1x_dota.py

AP50: 0.693636883638331
AP75: 0.34782717533485524
mAP: 0.3752184541769669

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ python run_net.py --config-file=configs/base.py --task=test
...
 |      KLD-R50-FPN       | DOTA1.0  |        1024/200         |      flip       |     -     |  SGD   |    1x    | 69.10  |                                                   [arxiv](https://arxiv.org/abs/2106.01883)                                                   |         [config](projects/kld/configs/rotated_retinanet_hbb_kld_r50_fpn_1x_dota.py)         |                                          [model](https://cloud.tsinghua.edu.cn/f/fa7e892f90304af6988b/?dl=1)                                           |
+|      KFIoU-R50-FPN     | DOTA1.0  |        1024/200         |      flip       |     -     |  SGD   |    1x    | 69.36  |                                                   [arxiv](https://arxiv.org/abs/2101.11952)|                                                            [config](projects/kfiou/configs/rotated_retinanet_hbb_kfiou_r50_fpn_1x_dota.py)                        |                                          [model](https://cloud.tsinghua.edu.cn/f/9d24118280864511b943/?dl=1) |
```
Comment (yangxue0827) [inline README.md:111]: Wrong arxiv link. 

KFIoU: https://arxiv.org/abs/2201.12558

File: projects/kfiou/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# KFIoU
+
+> [The KFIoU Loss for Rotated Object Detection](https://arxiv.org/pdf/2101.11952.pdf)
```
Comment (yangxue0827) [inline projects/kfiou/README.md:3]: Wrong arxiv link.

KFIoU: https://arxiv.org/abs/2201.12558

File: projects/kfiou/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+### Testing
+```sh
+python run_net.py --config-file=configs/rotated_retinanet_hbb_kfiou_r50_fpn_1x_dota.py --task=test
+```
+
+## Performance
+
+|   Models    | Dataset| Sub_Image_Size/Overlap |Train Aug | Test Aug | Optim | Lr schd | mAP    | Paper |                             Config                             | Download   |
+|:-----------:| :-----: |:-----:|:-----:| :-----: | :-----:| :-----:| :----: |:--------:|:--------------------------------------------------------------:| :--------: |
+| KFIoU-R50-FPN | DOTA1.0|1024/200| flip|-|  SGD   |   1x    | 69.36   | [arxiv](https://arxiv.org/abs/2101.11952)| [config](configs/rotated_retinanet_hbb_kfiou_r50_fpn_1x_dota.py) | [model](https://cloud.tsinghua.edu.cn/f/9d24118280864511b943/?dl=1) |
```
Comment (yangxue0827) [inline projects/kfiou/README.md:40]: Wrong arxiv link.

KFIoU: https://arxiv.org/abs/2201.12558

File: projects/kfiou/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Citation
+
+```
+@misc{yang2022kfiou,
```
Comment (yangxue0827) [inline projects/kfiou/README.md:45]:
```
@article{yang2022kfiou,
  title={The KFIoU Loss for Rotated Object Detection},
  author={Yang, Xue and Zhou, Yue and Zhang, Gefan and Yang, Jirui and Wang, Wentao and Yan, Junchi and Zhang, Xiaopeng and Tian, Qi},
  journal={arXiv preprint arXiv:2201.12558},
  year={2022}
}
```

File: projects/kfiou/test_rotated_retinanet_kfiou.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            losses = model(images,targets)
+            all_loss,losses = parse_losses(losses)
+            optimizer.step(all_loss)
+            correct_loss.append(all_loss.item())
+        data = {
+            "imagess": imagess,
+            "targetss": targetss,
+            "correct_loss": correct_loss,
+        }
+        if (not os.path.exists("test_datas_rotated_retinanet")):
```
Comment (yangxue0827) [inline projects/kfiou/test_rotated_retinanet_kfiou.py:47]: test_datas_rotated_retinanet -> test_datas_rotated_retinanet_kfiou

File: projects/kfiou/test_rotated_retinanet_kfiou.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            os.makedirs("test_datas_rotated_retinanet")
```
Comment (yangxue0827) [inline projects/kfiou/test_rotated_retinanet_kfiou.py:48]: test_datas_rotated_retinanet -> test_datas_rotated_retinanet_kfiou

File: projects/kfiou/test_rotated_retinanet_kfiou.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        pk.dump(data, open("test_datas_rotated_retinanet/test_data.pk", "wb"))
```
Comment (yangxue0827) [inline projects/kfiou/test_rotated_retinanet_kfiou.py:49]: test_datas_rotated_retinanet -> test_datas_rotated_retinanet_kfiou

File: projects/kfiou/test_rotated_retinanet_kfiou.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        print(correct_loss)
+    else:
+        data = pk.load(open("test_datas_rotated_retinanet/test_data.pk", "rb"))
```
Comment (yangxue0827) [inline projects/kfiou/test_rotated_retinanet_kfiou.py:52]: test_datas_rotated_retinanet -> test_datas_rotated_retinanet_kfiou

**Review by yangxue0827 — APPROVED** [review]
LGTM

**Review by zytx121 — APPROVED** [review]
LGTM
